### PR TITLE
Enabled async test at scala-native build

### PR DESCRIPTION
**Pull request by catap**
This commit is closing https://github.com/scalatest/scalatest/issues/1847

This commit working well on scala-native 0.4.0-M2 with Apple's clang `11.0.3 (clang-1103.0.32.62)` with unti tests that is using `wordspec.AsyncWordSpec`

I expect that everything will be fine on another backends.

Looks like this was disabled for a while ago, before 0.4.0-M2, and now it is safe assume that this tests is same to run ;)

### CODE REVIEW

**Comment by artimasites** [comment]
@cla-bot[bot] check

File: project/GenScalaTestNative.scala [hunk 1]
```diff
@@ -202,14 +194,14 @@ object GenScalaTestNative {
...
+      "org/scalatest/funspec" -> List.empty,
+      "org/scalatest/freespec" -> List.empty,
+      "org/scalatest/flatspec" -> List.empty,
       "org/scalatest/prop" -> List.empty, 
-      "org/scalatest/propspec" -> asyncs("jvm/propspec/src/main/scala/org/scalatest/propspec"), 
-      "org/scalatest/wordspec" -> asyncs("jvm/wordspec/src/main/scala/org/scalatest/wordspec"), 
+      "org/scalatest/propspec" -> List.empty,
+      "org/scalatest/wordspec" -> List.empty,
       "org/scalatest/concurrent" -> (List(
         "Waiters.scala",        // skipeed because doesn't really make sense on js's single-thread environment.
```
Comment (LeeTibbert) [inline project/GenScalaTestNative.scala:206]: s/skipeed/skipped/

Scalatest devos: Yes, please, this PR would make using Scalatest with Scala Native easier.
It appears to me to be Scala Native specific, so it should not perturb other platforms.

Thank you for considering it, especially with your busy workload.

Lee, an individual SN contributor, pecking away at hoped-for/eventual async support.

Comment (catap) [reply project/GenScalaTestNative.scala:206]: @LeeTibbert I just uncommented this lines, the same typo exists for example here: https://github.com/scalatest/scalatest/blob/f3d5f9d66631dc49b2218c8edd039840788bd915/project/GenScalaTestJS.scala#L225

**Comment by cheeseng** [comment]
@catap @LeeTibbert I tried this quickly on my machine (Ubuntu Linux), it seems async tests are failing: 

```
[info] *** 142 TESTS FAILED ***
[error] Failed tests:
[error]         org.scalatest.fixture.AsyncWordSpecSpec
[error]         org.scalatest.fixture.AsyncFunSuiteSpec
[error]         org.scalatest.fixture.AsyncFunSuiteLikeSpec
[error]         org.scalatest.AsyncWordSpecSpec
[error]         org.scalatest.fixture.DeprecatedAsyncFeatureSpecLikeSpec
[error]         org.scalatest.AsyncFunSuiteLikeSpec
[error]         org.scalatest.fixture.AsyncWordSpecLikeSpec
[error]         org.scalatest.fixture.DeprecatedAsyncFeatureSpecSpec
[error]         org.scalatest.AsyncFunSuiteSpec
[error]         org.scalatest.AsyncWordSpecLikeSpec
[error] (scalatestTestNative / Nativetest / test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 888 s (14:48), completed Jul 29, 2020 9:21:29 PM
```

I would like to understand why they are failing further, fyi here's how I built and ran them (I need to increase the max heap size to 10G after merging this PR, probably due to the increase of tests now): 

```
> sbt scalatestTestNative/clean scalacticTestNative/clean commonTestNative/clean scalatestNative/clean scalacticNative/clean
> sbt scalatestTestNative/test:compile
> sbt scalatestTestNative/test
```

Cheers.

**Comment by LeeTibbert** [comment]
@cheeseng  Thank you for having looked at this so soon.   Sorry that it failed for you.  I will build on my machine,
also Ubuntu (20.04). 

The current async case on Scala Native is a pain, so let me chase it.

I believe the intent is that the Fixture be available but any actual use of it beyond
an empty test should fail.    I will have to check that understanding with @catap.
My guess is that some tests should have been marked as skipped on Scala Native.

**Comment by catap** [comment]
@cheeseng let me check

**Comment by cheeseng** [comment]
@LeeTibbert @catap No worry, we're planning a dot release soon so it will be really exciting if we can have improvement on the scala-native support!  :)

Fyi I tried to enable Status's waitUntilCompleted, and when I tried to run it in one of the failed test it failed with: 

```
java.lang.UnsupportedOperationException: Scala.js is single-threaded!
[info]   at java.lang.Throwable.fillInStackTrace(Unknown Source)
[info]   at org.scalatest.CountDownLatch.await(Unknown Source)
[info]   at org.scalatest.CompositeStatus.waitUntilCompleted(Unknown Source)
[info]   at org.scalatest.AsyncWordSpecLikeSpec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(Unknown Source)
[info]   at org.scalatest.AsyncWordSpecLikeSpec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(Unknown Source)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(Unknown Source)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(Unknown Source)
[info]   at org.scalatest.Transformer.apply(Unknown Source)
[info]   at org.scalatest.Transformer.apply(Unknown Source)
[info]   at org.scalatest.funspec.AnyFunSpecLike$$anon$1.apply(Unknown Source)
```

It seems like it is some leftover code from scala-js.  I wonder in future scala-native probably should be able to support multithreading.  Anyway it is getting a bit late here, in case it is useful I have checked in my experiment branch here: 

https://github.com/cheeseng/scalatest/tree/scala-native-async-tests

**Comment by catap** [comment]
@cheeseng I'm testing it right now because I really would like to have enabled scala-native tests ;)

Anyway, I do have a few question regarding `.jvmopts`:
1) why do you force to use `MarkSweepGC`? Maybe remove this GC options. At jdk8 it will be the same GC, at something new it will be G1 or something better (=faster)
2) Can I increate `Xmx` to 10G or 16G as part of this pull request?

**Comment by cheeseng** [comment]
@catap @LeeTibbert And yes if it is a feature that not supported currently yet (e.g. latch), I think perhaps the best is to just skip them now, I think this PR has enabled all AsyncXXX tests.

**Comment by catap** [comment]
@cheeseng it seems that `ParallelTestExecution` doesn't work at scala-native and it is expected to be honest. 

Let me disable it as the next commit.

**Comment by cheeseng** [comment]
@catap 

>> 1. why do you force to use MarkSweepGC? Maybe remove this GC options. At jdk8 it will be the same GC, at something new it will be G1 or something better (=faster)

Ah yes, actually I am using export SBT_OPTS in my ~/.bashrc currently, which probably override the .jvmopts file, here's what I am currently using in muy ~/.bashrc: 

```
export SBT_OPTS="-server -Xms512M -Xmx10G -Xss10M -XX:+UseConcMarkSweepGC -XX:NewRatio=8"
```

>> 2. Can I increate Xmx to 10G or 16G as part of this pull request?

Sure, may scare some away from building scalatest though.  :D  Probably just the scala-native build needs it.  Anyway, I think machine with high RAM is getting common nowadays.  :)

**Comment by catap** [comment]
@cheeseng maybe remove `.jvmopts` at all? Modern JVM uses `1/4` of machine size for this value and it is *much* better than some hardcoded magic constant.

**Comment by catap** [comment]
@cheeseng I haven't run this by included unit test, but I run it on my project that also have quiet complicated tests.

But it isn't used overrided execution context. As soon as I removed all hacks for scala-native to override execution context, tests is passed.

So, let me add second commit to my pull request.

I haven't touched `.jvmopts` because I feel that it should be dedicated PR. And I feel that it should be removed, but I not sure how is it used and where. Just feeling.

**Comment by cheeseng** [comment]
@catap Yes I myself have not been using .jvmopts for a while, probably a separate issue, it should be good enough if we can get the async tests green for this PR, I'll try your new commit now.

Cheers.

**Comment by cheeseng** [comment]
@catap Tested your new commits and all is green.  I think it works because it will use the default serial execution context now which executes the Future right away, without needing to wait with waitUntilCompleted.

I think this is good to go into 3.2.x-new now, if @bvenners agrees and merge it into 3.2-x-new, I can backport it to 3.1.x branch.

**Comment by catap** [comment]
@cheeseng I've rebased https://github.com/catap/scalatest/tree/scala-native-async-tests to `3.1.x` branch but it doesn't pass tests in some strange place :(

I'm not sure that it is a good idea to invest more time to backport it to `3.1.x`. Honestly switch to 3.2 wasn't something difficult, and use this feature for force users to switch isn't such bad idea.

Anyway, if you think differently feel free to use it.

Maybe I will have some time to finish it but I don't promise anything.

**Comment by LeeTibbert** [comment]
@cheeseng @bvenners   Thank you for the timely merge.

**Pull request merged**